src/lib/components/DatePickerPanel.js

```diff
@@ -36,6 +36,12 @@ export default class DatePickerPanel extends React.Component {
     const { currentValue, confirmDisabled } = this.state;
     const disabledTimeFns = this.disabledTime();
 
+    const uiLocale = [browser.i18n.getUILanguage().replace('_', '-'), 'en-US'];
+    const dtf = new Intl.DateTimeFormat(uiLocale, {hour: 'numeric'});
+    const timeFormat = (dtf.resolvedOptions().hour12)
+      ? 'h:mm a'
+      : 'HH:MM';
+
```

Comment:
I believe you should. Not only because it'll do the hourCycle magic all by itself, but also because not all locales are left-to-right with western arabic numerals ;)
And Intl API handles that for you.